prereise/gather/griddata/transmission/geometry.py

```diff
@@ -0,0 +1,273 @@
+from dataclasses import dataclass
+from math import exp, log, pi, sqrt
+from statistics import geometric_mean
+
+from prereise.gather.griddata.transmission.const import (
+    epsilon_0,
+    mu_0,
+    relative_permeability,
+    resistivity,
+)
+
+
+@dataclass
+class Conductor:
+    """Represent a single conductor (which may be a stranded compsite).
+
+    :param float radius: outer radius of conductor.
+    :param str material: material of conductor. Used to calculate ``resistance_per_km``
+        and ``gmr`` if these aren't passed to the constructor, unnecessary otherwise.
+    :param float resistance_per_km: outer radius of conductor. Will be estimated from
+        other parameters if it isn't passed.
+    :param float gmr: geometric mean radius of conductor. Will be estimated from
+        other parameters if it isn't passed.
+    :param float area: cross-sectional area of conductor. Will be estimated from
+        other parameters if it isn't passed.
+    """
+
+    radius: float
+    material: str = None
+    resistance_per_km: float = None
+    gmr: float = None
+    area: float = None
```

Comment:
AFAIK, these variable types are annotations only, not sure whether we should perform type checks upon instantiation if it is important.

Reply:
You're right, if the user provides silly inputs they'll get an unintuitive error here:
```python
>>> from prereise.gather.griddata.transmission.geometry import Conductor
>>> c = Conductor(radius="foo", resistance_per_km="bar", material="aluminum")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "<string>", line 8, in __init__
  File "C:\Users\DanielOlsen\repos\bes\PreREISE\prereise\gather\griddata\transmission\geometry.py", line 52, in __post_init__
    self.gmr = self.radius * exp(self.permeability / 4)
TypeError: can't multiply sequence by non-int of type 'float'
```

I added a parent class with a validation method, so now instead we get:
```python
>>> c = Conductor(radius="foo", resistance_per_km="bar", material="aluminum")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "<string>", line 8, in __init__
  File "C:\Users\DanielOlsen\repos\bes\PreREISE\prereise\gather\griddata\transmission\geometry.py", line 46, in __post_init__
    self._validate_input_types()
  File "C:\Users\DanielOlsen\repos\bes\PreREISE\prereise\gather\griddata\transmission\geometry.py", line 18, in _validate_input_types
    raise TypeError(
TypeError: for radius: expected <class 'float'>, but got <class 'str'>
```

Reply:
Okay, the updated code does the trick.